web/src/upload_widget.ts

```diff
@@ -135,61 +468,126 @@ export function build_direct_upload_widget(
     // jQuery button to open file dialog
     $upload_button: JQuery,
     upload_function: UploadFunction,
+    // default value of max uploaded file size
     max_file_upload_size: number,
+    // New profile picture or New organization logo or New organization icon
+    heading: string,
+    // Cropper options passed by the caller
+    cropper_options: CropperOptions,
 ): void {
-    // default value of max uploaded file size
-    function accept(): void {
-        $input_error.hide();
-        const $realm_logo_section = $upload_button.closest(".image_upload_widget");
-        if ($realm_logo_section.attr("id") === "realm-night-logo-upload-widget") {
-            upload_function(get_file_input(), true, false);
-        } else if ($realm_logo_section.attr("id") === "realm-day-logo-upload-widget") {
-            upload_function(get_file_input(), false, false);
-        } else {
-            upload_function(get_file_input(), null, true);
-        }
+    let first = true;
+    function submit(): void {
+        edited = false;
+        $(".uppy-DashboardContent-save").trigger("click");
     }
 
     function clear(): void {
+        if (uppy) {
+            uppy.close();
+        }
         const $control = get_file_input();
         $control.val("");
+        first = true;
+    }
+
+    function accept(file: File, scalable: boolean): void {
+        $input_error.hide();
+        if (first) {
+            first = false;
+            dialog_widget.launch({
+                html_heading: heading,
+                html_body: "<div id='ImageEditorDiv'></div>",
+                html_submit_button: `${$t_html({
+                    defaultMessage: "Save changes",
+                })} <button class='modal__btn hide direct_widget_save' id='scale_to_fit'>Scale to fit</button> <button class='modal__btn hide direct_widget_save' id='reset_scale_to_fit'>Reset scale</button>`,
```

Comment:
It doesn't look so good to put these buttons here. Any idea how buttons other than Save and Cancel is handled in other modals?

Reply:
I'm not able to find a modal with more than 2 buttons other than the ones that we built where we simply add the buttons in the respective `.hbs` file. I think adding these two new buttons directly in the `dialog_widget.hbs` is not the best solution. What do you think?